src/app/loginComponents/accounts/internal/change-username/change-username.component.ts

```diff
@@ -97,4 +98,17 @@ export class ChangeUsernameComponent implements OnInit {
   updateUsername() {
     this.userService.updateUsername(this.username);
   }
+
+  /**
+   * Enable or disable form input if user can or cannot update username
+   */
+  enableDisableFormControl() {
+    this.canChangeUsername$.subscribe((canChangeUsername: boolean) => {
```

Comment:
Not sure, but I think you need to unsubscribe.

Or better, I think, just use `this.canChangeUsername$` directly in HTML like the button does [here](https://github.com/dockstore/dockstore-ui2/blob/ed75de98a8018ae72348a30de59da8c08a3ef05b/src/app/loginComponents/accounts/internal/change-username/change-username.component.html#L48-L47) (I think you can just add disabled to the input).

Reply:
> Yeah I first tried disabling it from the html directly, but for reactive forms it's not possible

Thanks for clarifying, I did not know that.

> I'll update this to unsubscribe?

I think you need to, yes.  And I think the way to it is to use `takeUntil`, like line 61.